contracts/Backing.sol

```diff
@@ -0,0 +1,154 @@
+        require(assets[weth].target != address(0), "weth has not been registed");
+        IWETH(weth).deposit{value: msg.value}();
+        emit BackingLock(weth, assets[weth].target, msg.value, recipient);
+    }
+
+    function crossChainSync(
+        bytes memory message,
+        bytes[] memory signatures,
+        bytes32 root,
+        uint32 MMRIndex,
+        bytes memory blockHeader,
+        bytes32[] memory peaks,
+        bytes32[] memory siblings,
+        bytes memory eventsProofStr
+    ) public {
+        if(relay.getMMRRoot(MMRIndex) == bytes32(0)) {
+            relay.appendRoot(message, signatures);
+        }
+        verifyProof(root, MMRIndex, blockHeader, peaks, siblings, eventsProofStr);
+    }
+
+    function verifyProof(
+        bytes32 root,
+        uint32 MMRIndex,
+        bytes memory blockHeader,
+        bytes32[] memory peaks,
+        bytes32[] memory siblings,
+        bytes memory eventsProofStr
+    ) public {
+        uint32 blockNumber = Scale.decodeBlockNumberFromBlockHeader(blockHeader);
+
+        require(history[blockNumber] == address(0), "TokenBacking:: verifyProof:  The block has been verified");
+
+        Input.Data memory data = Input.from(relay.verifyRootAndDecodeReceipt(root, MMRIndex, blockNumber, blockHeader, peaks, siblings, eventsProofStr, storageKey));
+        
+        ScaleStruct.IssuingEvent[] memory events = Scale.decodeIssuingEvent(data);
+
+        uint256 len = events.length;
+        for( uint i = 0; i < len; i++ ) {
+          ScaleStruct.IssuingEvent memory item = events[i];
+          if (item.eventType == 1) {
+              processRedeemEvent(item);
+          } else if (item.eventType == 0) {
+              processRegisterResponse(item);
+          }
+        }
+
+        history[blockNumber] = msg.sender;
+        emit VerifyProof(blockNumber);
+    }
+
+    function processRedeemEvent(ScaleStruct.IssuingEvent memory item) internal {
+        uint256 value = item.value;
+        address token = item.token;
+        address target = item.target;
+        require(assets[token].target == target, "the mapped address uncorrect");
+        require(item.backing == address(this), "not the expected backing");
+        // assetType == 0: native, 1: token
+        if (token == weth && item.assetType == 0) {
+            IWETH(weth).withdraw(value);
+            item.recipient.transfer(value);
+        } else {
+            IERC20(token).safeTransfer(item.recipient, value);
+        }
+        emit RedeemTokenEvent(token, target, item.recipient, value);
+    }
+
+    function processRegisterResponse(ScaleStruct.IssuingEvent memory item) internal {
+        address token = item.token;
+        require(assets[token].timestamp != 0, "asset is not existed");
+        require(assets[token].target == address(0), "asset has been responsed");
+        require(item.backing == address(this), "not the expected backing");
+        address target = item.target;
```

Comment:
What's the target?

The mapping token address on the target chain?

Reply:
yes